orchestrator/cli/migrate_workflows.py

```diff
@@ -0,0 +1,237 @@
+# We can assume that production environments are always test envs in terms of migrations executed. It is therefore ok use workflow names in the migration files, since
+# the database should only contain workflow names added from previous migration files.
+
+# Workflows without a product associated (tasks) should not be removed
+
+logger = structlog.get_logger(__name__)
+
+T = TypeVar("T")
+
+
+def _enumerate_menu_keys(items: list) -> List[str]:
+    return [str(i + 1) for i in range(len(items))]
+
+
+def _prompt_user_menu(
+    options: Iterable[Tuple[str, T]], keys: Optional[List[str]] = None, repeat: bool = True
+) -> Optional[T]:
+    options_list = list(options)
+    keys = keys or _enumerate_menu_keys(options_list)
+    done = False
+    while not done:
+        for k, txt_v in zip(keys, options_list):
+            noqa_print(f"{k}) {txt_v[0]}")
+        choice = get_user_input("? ")
+        if choice not in keys:
+            noqa_print("Invalid choice")
+            done = not repeat
+        else:
+            return options_list[keys.index(choice)][1]
+    return None
+
+
+def _print_workflows_table(workflows: List[WorkflowTable]) -> None:
+    items = [(wf.name, wf.target, wf.description, wf.products[0].product_type) for wf in workflows if wf.products]
+    print_fmt(
+        tabulate(
+            items,
+            headers=["#", "name", "target", "description", "product_type"],
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+
+
+def _add_workflow(workflows: Dict[str, LazyWorkflowInstance], state: dict) -> dict:
+    print_fmt("\nAdd new workflow\n", flags=[COLOR.UNDERLINE])
+
+    if not workflows:
+        noqa_print("No registered workflows found to add to the database")
+        return state
+
+    registered_product_types = [
+        row[0] for row in ProductTable.query.with_entities(ProductTable.product_type).distinct()
+    ]
+    if not registered_product_types:
+        noqa_print("No registered product types found")
+        return state
+
+    noqa_print("Which product type should the workflow be added to?")
+    product_type = _prompt_user_menu(
+        [*[(p, p) for p in registered_product_types], ("cancel", None)],
+        keys=[*_enumerate_menu_keys(registered_product_types), "q"],
+    )
+
+    if not product_type:
+        # Menu cancelled
+        return state
+
+    noqa_print(f"\nAdd product type {str_fmt(product_type, flags=[COLOR.BOLD])} to which workflow?")
+
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    wf_options = [(wf, wf) for wf in workflows if wf not in already_used_workflows]
+    wf_name = _prompt_user_menu([*wf_options, ("cancel", None)], keys=[*_enumerate_menu_keys(wf_options), "q"])
+    if not wf_name:
+        # Menu cancelled
+        return state
+
+    wf_inst = get_workflow(wf_name)
+    if wf_inst is None:
```

Comment:
Nitpick: can save a line with the walrus :)

```suggestion
    if not wf_inst := get_workflow(wf_name):
```